2018/python/file_generator.py

```diff
@@ -487,7 +487,115 @@ def edit_report(self):
 			filepath=edit_report_path))
 
 
+	def create_custom_row(self, dictionary, clean_filepath, clean_filename):
 
+		"""
+		Creates a custom clean LAR row by passing in a dictionary of columns 
+		and new values to modify all the rows of an 
+		existing clean file, filters the modified file for clean rows, 
+		and then pulls the first row from the file. 
+		Pulls rows from the clean file last generated. Suggestion that 
+		the file pulled should be 1000 original rows or greater, to ensure
+		that modified clean rows can be found. 
+		"""
+		#Creates a TS and LAR dataframe from the clean filepath and name
+		#specified. 
+		ts_df, lar_df = utils.read_data_file(
+					path=clean_filepath,
+					data_file=clean_filename)
+
+		#Changes each column (key in the dictionary) to the new value in 
+		# the dictionary.
+		for key, value in dictionary.items():
+			lar_df[key] = value
+
+		checker = rules_engine(lar_schema=self.lar_schema_df, 
+					ts_schema=self.ts_schema_df, 
+					crosswalk_data=self.crosswalk_data)
+
+		#Produces a report as to which syntax or validity
+		#edits have passed or failed based on logic in the rules_engine.
+		#Loads the TS and LAR dataframes into the checker object. 
+
+		checker.load_data_frames(ts_df, lar_df)
+
+		for func in dir(checker):
+			if func[:1] in ("s", "v") and func[1:4].isdigit()==True:
+				getattr(checker, func)()
+
+	
+		#Produces a report as to which syntax or validity
+		#edits have passed or failed based on logic in the rules_engine.
+		for func in dir(checker):
+			if func[:1] in ("s", "v") and func[1:4].isdigit()==True:
+				getattr(checker, func)()
+		
+		#Creates a results dataframe and keeps the results that 
+		#have failed. 
+		res_df = pd.DataFrame(checker.results)
+		new_df = res_df[(res_df['status']=='failed')]
```

Comment:
Please rename new_df to indicate its purpose

Reply:
using `res_df = res_df[(res_df['status']=='failed')].copy()` will remove a memory object you're not using.

Reply:
I will use that! Thank you.